src/components/Reports/ReportsHoursVisits/ReportsHoursVisits.test.js

```diff
@@ -0,0 +1,162 @@
+          to: date.toString(),
+        },
+      ],
+      qVisitors: Math.floor(Math.random() * 1000),
+      qVisitorsWithEmail: 1,
+      qVisits: Math.floor(Math.random() * 1000 + 1000),
+      qVisitsWithEmail: Math.floor(Math.random() * 100),
+    };
+  });
+};
+
+describe('reports weekday and hours visits', () => {
+  afterEach(cleanup);
+
+  it('should deal with DataHub failure', async () => {
+    // Arrange
+    const dataHubClientDouble = {
+      getVisitsQuantitySummarizedByWeekdayAndHour: async () => errorResponse,
+    };
+
+    // Act
+    const { container, getByText } = render(
+      <AppServicesProvider
+        forcedServices={{
+          datahubClient: dataHubClientDouble,
+        }}
+      >
+        <DopplerIntlProvider locale="es">
+          <ReportsHoursVisits domainName={domainName} dateFrom={dateFrom} dateTo={dateTo} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(container.querySelector('.loading-box')).toBeInTheDocument();
+    await waitFor(() => expect(getByText('common.unexpected_error')));
+  });
+
+  it('should show the graphic with the data', async () => {
+    // Arrange
+    const dataHubClientDouble = {
+      getVisitsQuantitySummarizedByWeekdayAndHour: async () => getFakeHoursVisitsData(),
+    };
+
+    // Act
+    const { container } = render(
+      <AppServicesProvider
+        forcedServices={{
+          datahubClient: dataHubClientDouble,
+        }}
+      >
+        <DopplerIntlProvider locale="en">
+          <ReportsHoursVisits domainName={domainName} dateFrom={dateFrom} dateTo={dateTo} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(container.querySelector('.loading-box')).toBeInTheDocument();
+    await waitFor(() => {
+      expect(container.querySelector('.loading-box')).not.toBeInTheDocument();
+    });
+  });
+
+  it('should show the graphic and check specific data', async () => {
+    // Arrange
+    const fakeHoursVisits = {
+      success: true,
+      value: [
+        {
+          weekday: 0,
+          hour: 0,
+          qVisitors: 593,
+          qVisitorsWithEmail: 0,
+          qVisitorsWithOutEmail: 0,
+          qVisits: 800,
+          qVisitsWithEmail: 100,
+        },
+      ],
+    };
+
+    const dataHubClientDouble = {
+      getVisitsQuantitySummarizedByWeekdayAndHour: async () => fakeHoursVisits,
+    };
+
+    // Act
+    const { container, getByText } = render(
+      <AppServicesProvider
+        forcedServices={{
+          datahubClient: dataHubClientDouble,
+        }}
+      >
+        <DopplerIntlProvider locale="en">
+          <ReportsHoursVisits domainName={domainName} dateFrom={dateFrom} dateTo={dateTo} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(container.querySelector('.loading-box')).toBeInTheDocument();
+    await waitFor(() => expect(getByText('593')));
```

Comment:
Perhaps this `593` could be in a variable.

Reply:
we have before this discussion and says the both options are ok